geomstats/information_geometry/multivariate_normal.py

```diff
@@ -9,13 +9,137 @@
 
 import geomstats
 import geomstats.backend as gs
+import geomstats.errors as errors
 from geomstats.geometry.base import OpenSet
 from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.product_manifold import ProductManifold
+from geomstats.geometry.product_riemannian_metric import NFoldMetric
 from geomstats.geometry.riemannian_metric import RiemannianMetric
+from geomstats.geometry.spd_matrices import SPDAffineMetric, SPDMatrices
 from geomstats.information_geometry.base import InformationManifoldMixin
 from geomstats.information_geometry.normal import NormalMetric
 
 
+class MultivariateNormalDistributions:
+    """Class for the multivariate normal distributions.
+
+    This class is a common interface to the following different situations:
+
+    - centered distributions
+    - distributions with diagonal covariance matrix
+    - general distributions
+
+    Parameters
+    ----------
+    n : int
+        Dimension of the sample space of the multivariate normal distribution.
+    distribution_type : str, {'centered', 'diagonal', 'general'}
+        Type of distributions.
+        Optional, default: 'general'.
+    """
+
+    def __new__(cls, n, distribution_type="general"):
+        """Instantiate class that corresponds to the distribution_type."""
+        errors.check_parameter_accepted_values(
+            distribution_type,
+            "distribution_type",
+            ["centered", "diagonal", "general"],
+        )
+        if distribution_type == "centered":
+            return MultivariateCenteredNormalDistributions(n)
+        if distribution_type == "diagonal":
+            return MultivariateDiagonalNormalDistributions(n)
+        return MultivariateGeneralNormalDistributions(n)
+
+
+class MultivariateCenteredNormalDistributions(InformationManifoldMixin, SPDMatrices):
+    """Class for the manifold of centered multivariate normal distributions.
+
+    This is the class for multivariate normal distributions with zero mean
+    on the $n$-dimensional Euclidean space. Each distribution is represented by
+    its covariance matrix, i.e. an $n$-by-$n$ symmetric positive-definite matrix.
+
+    Parameters
+    ----------
+    n : int
```

Comment:
That might be a remark for a few of the information manifolds: do we want to rename this `n` as `sample_dim` or another more explicit variable name? 

Pros:
- since `n` is used in statistics to denote the sample size, n as the dimension of the sample space can be confusing.

Cons:
- `n` here also corresponds to the `n` in SPDMatrices, thus that part is nice.

I do not have a strong opinion on this, we can keep it like this for now :)